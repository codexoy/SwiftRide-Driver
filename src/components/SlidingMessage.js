import React from "react"
import { StyleSheet, View, Text } from "react-native"
import Modal from "react-native-modal"
import { errors } from "../i18n"

export default class SlidingMessage extends React.Component {
	render() {
		let color = "green"
		if (this.props.error) {
			color = "red"
		}
		return (
			<Modal
				isVisible={this.props.visible}
				animationIn="slideInDown"
				animationOut="slideOutUp"
				style={styles.modal}
				hasBackdrop={true}
				backdropOpacity={0}
				onBackdropPress={this.props.close}
				swipeThreshold={10}
				swipeDirection="up"
				onSwipeComplete={this.props.close}
			>
				<View
					style={{
						...styles.view,
						backgroundColor: color
					}}
				>
					<Text style={styles.text}>
						{this.props.error
							? errors(`errors.${this.props.error}`)
							: this.props.success}
					</Text>
				</View>
			</Modal>
		)
	}
}

const styles = StyleSheet.create({
	modal: {
		justifyContent: "flex-start",
		margin: 0
	},
	view: {
		height: 100,
		padding: 20,
		paddingTop: 60,
		alignItems: "flex-start"
	},
	text: {
		color: "#fff"
	}
})
