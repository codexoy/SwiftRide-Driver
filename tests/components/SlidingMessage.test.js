import React from "react"

import renderer from "react-test-renderer"

import SlidingMessage from "../../src/components/SlidingMessage"

describe("SlidingMessage", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<SlidingMessage
					success="success!"
					visible={true}
					close={jest.fn}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
