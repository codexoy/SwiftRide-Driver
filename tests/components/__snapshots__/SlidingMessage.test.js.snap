// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlidingMessage view renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  onSwipeComplete={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeDirection="up"
  swipeThreshold={10}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeComplete={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "left": 0,
        "margin": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeDirection="up"
    swipeThreshold={10}
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "green",
          "height": 100,
          "padding": 20,
          "paddingTop": 60,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
          }
        }
      >
        success!
      </Text>
    </View>
  </View>
</Modal>
`;
